packages/core/upload/server/controllers/validation/admin/upload.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const { yup, validateYupSchema } = require('@strapi/utils');
+const { isNil } = require('lodash/fp');
+const { getService } = require('../../../utils');
+
+const fileInfoSchema = yup.object({
+  name: yup.string().nullable(),
+  alternativeText: yup.string().nullable(),
+  caption: yup.string().nullable(),
+  folder: yup
+    .strapiID()
+    .nullable()
+    .test('folder-exists', "the folder doesn't exist", async folderId => {
+      if (isNil(folderId)) {
+        return true;
+      }
+
+      return getService('folder').exists({ id: folderId });
```

Comment:
Is this missing an `await`?

Reply:
Yeah, I'm asking because `.exists()` returns a different `Promise`, which might lead to unexpected behavior ... Everywhere else you are awaiting before returning, which I'd prefer here too:

```suggestion
      const exists = await getService('folder').exists({ id: folderId });
      
      return exists;
```

Reply:
I'm not sure to understand really why. However why not, will do that :)